src/frontend/components/PublicLiveDashboard/index.tsx

```diff
@@ -22,14 +25,30 @@ export const PublicLiveDashboard = ({
 }: PublicLiveDashboardProps) => {
   const intl = useIntl();
   const [isReadyLive, setIsReadyLive] = useState(false);
-
-  useAsyncEffect(async () => {
-    if (!isReadyLive) {
-      await initWebinarContext(live, intl.locale);
-      initVideoWebsocket(live);
-      setIsReadyLive(true);
-    }
-  }, [isReadyLive, live]);
+  const anonymousId = useMemo(() => getOrInitAnonymousId(), []);
+  const queryClient = useQueryClient();
+  const setLiveSession = useLiveSession((state) => state.setLiveSession);
+  useLiveSessions(
+    { anonymous_id: anonymousId },
+    {
+      onSuccess: async (data) => {
+        if (data.count > 0) {
+          setLiveSession(data.results[0]);
+        } else {
+          setLiveSession(await pushAttendance({}, intl.locale, anonymousId));
+          queryClient.invalidateQueries('livesessions', {
+            refetchActive: false,
+          });
+        }
+        initVideoWebsocket(live);
+        setIsReadyLive(true);
```

Comment:
This state might be useless now (since as soon as we have a response from react-query even when loading it means the live is ready). 
I'm just pointing this out, I think it might be better to keep the state to get a more explicit (and you will probably need it if you define `onError`).